src/o365/teams/commands/report/teams-report-deviceusagedistributionusercounts.spec.ts

```diff
@@ -124,7 +168,140 @@ describe(commands.TEAMS_REPORT_DEVICEUSAGEDISTRIBUTIONUSERCOUNTS, () => {
         done(e);
       }
     });
+  });
+
+  it('gets the number of Microsoft Teams unique users by device type for the given period and export report data in txt format', (done) => {
+    const requestStub: sinon.SinonStub = sinon.stub(request, 'get').callsFake((opts) => {
+      if (opts.url === `https://graph.microsoft.com/v1.0/reports/getTeamsDeviceUsageDistributionUserCounts(period='D7')`) {
+        return Promise.resolve(`
+        Report Refresh Date,Web,Windows Phone,Android Phone,iOS,Mac,Windows,Report Period
+        2019-08-28,0,0,0,0,0,0,7
+        `);
+      }
+
+      return Promise.reject('Invalid request');
+    });
+
+    const fileStub: sinon.SinonStub = sinon.stub(fs, 'writeFileSync').callsFake(writeFileSyncFake);
+
+    cmdInstance.action({ options: { debug: false, period: 'D7', outputFilePath: '/Users/josephvelliah/Desktop/deviceusagedistributionusercounts.txt' } }, () => {
+      try {
+        assert.equal(requestStub.lastCall.args[0].url, "https://graph.microsoft.com/v1.0/reports/getTeamsDeviceUsageDistributionUserCounts(period='D7')");
```

Comment:
Let's try to limit tests to one assert per test. With multiple asserts, when a test fails, it's hard to figure out which one failed. If there are multiple things to assess, let's put them in separate tests.

Reply:
This test case was added by VelinGeorgiev with this approach. Could you please let me or show me how to change this? Please feel free to update this section if it is simple.